harness/determined/common/context.py

```diff
@@ -66,34 +66,60 @@ def v1File_from_local_dir(path: str, local_path: pathlib.Path) -> bindings.v1Fil
 
 
 def read_v1_context(
-    local_path: pathlib.Path,
+    root_path: pathlib.Path,
     limit: int = constants.MAX_CONTEXT_SIZE,
 ) -> List[bindings.v1File]:
     """
     Return a list of v1Files suitable for submitting a context directory over the v1 REST API.
 
+    If root_path refers to a file, a context directory is created containing only that file.
+
     A .detignore file in the directory, if specified, indicates the wildcard paths
     that should be ignored. File paths are represented as relative paths (relative to
     the root directory).
     """
 
+    root_path = root_path.resolve()
+
+    if not root_path.exists():
+        raise Exception(f"Path '{root_path}' doesn't exist")
+
     items = []
     size = 0
 
     def add_item(f: bindings.v1File) -> None:
         nonlocal size
         items.append(f)
         size += v1File_size(f)
+        if size > limit:
+            print()
+            raise ValueError(
+                f"Directory '{root_path}' exceeds the maximum allowed size "
+                f"{sizeof_fmt(constants.MAX_CONTEXT_SIZE)}.\n"
+                "Consider using a .detignore file to specify that certain files "
+                "or directories should be omitted from the context directory."
+            )
 
-    local_path = local_path.resolve()
+    msg = f"Preparing files (in {root_path}) to send to master... {sizeof_fmt(0)} and {0} files"
+    print(msg, end="\r", flush=True)
 
-    if not local_path.exists():
-        raise Exception(f"Path '{local_path}' doesn't exist")
+    def update_msg() -> None:
+        nonlocal msg
+        print(" " * len(msg), end="\r")
+        msg = (
+            f"Preparing files ({root_path}) to send to master... "
+            f"{sizeof_fmt(size)} and {len(items)} files"
+        )
+        print(msg, end="\r", flush=True)
 
-    if local_path.is_file():
-        raise ValueError(f"Path '{local_path}' must be a directory")
+    if root_path.is_file():
+        # Single-file context case.
+        add_item(v1File_from_local_file(root_path.name, root_path))
+        update_msg()
+        print()
+        return items
 
-    root_path = local_path
+    root_path = root_path
```

Comment:
find-replace?

Reply:
Yes... why accept a badly named parameter, then rename it halfway through the function without modification?  So I fixed it.

Reply:
ah, but I see the mistake here, thank you.